src/base-command.ts

```diff
@@ -29,6 +39,110 @@ export default abstract class BaseCommand extends Command {
+      ...Object.entries(calling_class.flags || {}).filter(([_, value]) => (value as any).non_sensitive).map(([key, _]) => key),
+      ...Object.entries(calling_class.args || {}).filter(([_, value]) => (value as any).non_sensitive).map(([_, value]) => (value as any).name),
+    ]);
+
+    const { args, flags } = await this.parse(calling_class);
+
+    const filtered_sentry_args = await this._filterNonSensitiveSentryMetadata(non_sensitive, _args || args);
+    const filtered_sentry_flags = await this._filterNonSensitiveSentryMetadata(non_sensitive, _flags || flags);
+
+    return { filtered_sentry_args, filtered_sentry_flags };
+  }
+
+  async setupAnalytics(): Promise<void> {
+    const analytics_env = this.oclif_env;
+    Sentry.init({
+      dsn: CLI_SENTRY_DSN,
+      debug: false,
+      environment: analytics_env,
+      release: process.env?.npm_package_version,
+      tracesSampleRate: 1.0,
+      attachStacktrace: true,
+      sendClientReports: true,
+      integrations: [
+        new Dedupe(),
+        new RewriteFrames({
+          root: LocalPaths.SENTRY_ROOT_PATH,
+          prefix: `src/`,
+        }),
+        new ExtraErrorData(),
+        new Transaction(),
+      ],
+      beforeSend(event: any) {
+        if (analytics_env === 'local') return null;
+        if (event.req?.data?.token) {
+          event.req.data.token = '*'.repeat(20);
+        }
+        return event;
+      },
+    });
+
+    const auth_user = await this.app?.auth?.getPersistedTokenJSON();
+    const auth_login = await this.app?.auth?.checkLogin();
+
+    const sentry_session_tags = {
+      environment: analytics_env,
+      cli: this.app.version,
+      node_runtime: process.version,
+      os: os.platform(),
+      shell: this.config?.shell,
+      user: auth_login?.id || os.hostname(),
+      'user-email': auth_user?.email || os.hostname(),
+    };
+
+    const sentry_session_user = {
+      email: auth_user?.email || os.hostname(),
+      id: auth_login?.id || os.hostname(),
+    };
+
+    const docker_version = await docker(['version', '-f', 'json'], { stdout: false });
+    const docker_containers = await docker(['ps', '--format', '{"ID":"{{ .ID }}", "Image": "{{ .Image }}", "Name":"{{ .Names }}"},'], { stdout: false });
+
+    const sentry_session_docker_containers = JSON.parse(`[${(docker_containers.stdout as string)
+      .substring(0, (docker_containers.stdout as string).length - 1)}]`);
+    const sentry_session_docker_version = JSON.parse(docker_version.stdout as string);
+
+    const sentry_session_metadata = {
+      command: this.id || (this.constructor as any).name,
+      environment: analytics_env,
+      email: auth_user?.email || '',
+      id: auth_login?.id || os.hostname(),
+      config_file: path.join(this.app?.config?.getConfigDir(), LocalPaths.CLI_CONFIG_FILENAME),
+      cwd: process.cwd(),
+      docker_info: sentry_session_docker_version,
+      docker_containers: sentry_session_docker_containers,
```

Comment:
Won't this only be set from when the command starts? Basically fi there is an error in the middle of the command and that command spun up some containers, we would not be able to see the new containers. This needs to happen when the error occurs and NOT on startup. The same goes for `linked_components`

Reply:
moved setting of metadata to occur at time of error during the catch function.